crates/burn-import/src/burn/node/split.rs

```diff
@@ -0,0 +1,137 @@
+    }
+
+    fn input_types(&self) -> Vec<Type> {
+        vec![Type::Tensor(self.input.clone())]
+    }
+
+    fn forward(&self, scope: &mut Scope, node_position: usize) -> TokenStream {
+        let input = scope.tensor_use_owned(&self.input, node_position);
+        let axis = self.config.axis.to_tokens();
+        let split_tensors = syn::Ident::new("split_tensors", proc_macro2::Span::call_site());
+
+        let split_code = if let Some(split_sizes) = &self.config.split_sizes {
+            let split_sizes_tokens = split_sizes.to_tokens();
+            quote! {
+                let #split_tensors = #input.split_with_sizes(#split_sizes_tokens, #axis);
+            }
+        } else {
+            let num_outputs = self.config.num_outputs.unwrap();
+            let num_outputs_tokens = num_outputs.to_tokens();
+            quote! {
+                let #split_tensors = #input.split(#num_outputs_tokens, #axis);
+            }
+        };
+
+        split_code
+    }
+
+    fn into_node(self) -> Node<PS> {
+        Node::Split(self)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use burn::record::FullPrecisionSettings;
+
+    use super::*;
+    use crate::burn::{
+        graph::BurnGraph,
+        node::{split::SplitNode, test::assert_tokens},
+        TensorType,
+    };
+
+    #[test]
+    fn test_codegen_split() {
+        let mut graph = BurnGraph::<FullPrecisionSettings>::default();
+
+        graph.register(SplitNode::new(
+            TensorType::new_float("tensor1", 2),
+            vec![
+                TensorType::new_float("tensor2", 2),
+                TensorType::new_float("tensor3", 2),
+            ],
+            SplitConfig {
+                axis: 0,
+                num_outputs: Some(2),
+                split_sizes: None,
+            },
+        ));
+
+        graph.register_input_output(
+            vec!["tensor1".to_string()],
+            vec!["split_tensors".to_string()],
+        );
+
+        let expected = quote! {
+            use burn::{
+                module::Module,
+                tensor::{backend::Backend, Tensor},
+            };
+
+            #[derive(Module, Debug)]
+            pub struct Model<B: Backend> {
+                phantom: core::marker::PhantomData<B>,
+                device: burn::module::Ignored<B::Device>,
+            }
+
+            impl<B: Backend> Model <B> {
+                #[allow(unused_variables)]
+                pub fn new(device: &B::Device) -> Self {
+                    Self {
+                        phantom: core::marker::PhantomData,
+                        device: burn::module::Ignored(device.clone()),
+                    }
+                }
+
+                #[allow(clippy::let_and_return, clippy::approx_constant)]
+                pub fn forward(
+                    &self,
+                    tensor1: Tensor<B, 2>,
+                ) -> Vec<Tensor<B, 2>> {
+                    let split_tensors = tensor1.split(2, 0);
+
+                    split_tensors
```

Comment:
And so the generated function should look like this instead:
```suggestion
                ) -> (Tensor<B, 2>, Tensor<B, 2>) {
                    let split_tensors = tensor1.split(2, 0);
                    let [tensor2, tensor3] = split_tensors.try_into().unwrap();
                    (tensor2, tensor3)
```